mathesar/tests/views/api/test_table_api.py

```diff
@@ -873,3 +885,75 @@ def test_table_viewset_checks_cache(client):
     with patch.object(reflection, 'reflect_tables_from_schema') as mock_reflect:
         client.get('/api/v0/tables/')
     mock_reflect.assert_called()
+
+
+def test_table_retrieve_no_primary_key(create_table, client):
+    table_name = 'NASA Table Retrieve'
+    new_table_name = 'NASA Table Retrieve New'
+    table = create_table(table_name)
+
+    body = {'name': new_table_name, 'num_primary_keys': 0}
```

Comment:
`get` requests do not take a body.

Also, you shouldn't be setting the `num_primary_keys` value, it should be read only. You need to verify that the value returned for the table is `0`.

Reply:
Verifying the `num_primary_keys` value should be in a separate test from retrieve/delete/partial update tests.